src/Nethermind/Nethermind.DataMarketplace.Consumers.Test/Services/DepositUnitsCalculatorTests.cs

```diff
@@ -0,0 +1,125 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Test.Builders;
+using Nethermind.DataMarketplace.Consumers.Deposits;
+using Nethermind.DataMarketplace.Consumers.Deposits.Domain;
+using Nethermind.DataMarketplace.Consumers.Deposits.Services;
+using Nethermind.DataMarketplace.Consumers.Sessions.Domain;
+using Nethermind.DataMarketplace.Consumers.Sessions.Queries;
+using Nethermind.DataMarketplace.Consumers.Sessions.Repositories;
+using Nethermind.DataMarketplace.Core.Domain;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace Nethermind.DataMarketplace.Consumers.Test.Services
+{
+    public class DepositUnitsCalculatorTests
+    {
+        private IConsumerSessionRepository _sessionRepository;
+        private ITimestamper _timestamper;
+        private IDepositUnitsCalculator _calculator;
+        private const uint DepositConfirmationTimestamp = 1546300800;
+        private static readonly DateTime Date = new DateTime(2019, 1, 2); //1546383600
+
+        [SetUp]
+        public void Setup()
+        {
+            _sessionRepository = Substitute.For<IConsumerSessionRepository>();
+            _timestamper = new Timestamper(Date);
+            _calculator = new DepositUnitsCalculator(_sessionRepository, _timestamper);
+        }
+
+        [Test]
+        public async Task should_return_0_consumed_units_for_unconfirmed_deposit()
+        {
+            var deposit = GetDepositDetails(confirmationTimestamp: 0);
+            var consumedUnits = await _calculator.GetConsumedAsync(deposit);
+
+            consumedUnits.Should().Be(0);
+
+            await _sessionRepository.DidNotReceive().BrowseAsync(Arg.Any<GetConsumerSessions>());
+        }
+
+        [Test]
```

Comment:
for the future it would be good to see if exceptions are handled properly - what if the sessions total somehow ends up being above what was allowed? would there be a valid error log / would we prevent the error from propagating in the system?

Reply:
Good point.